app/src/main/java/com/nytimes/android/sample/SampleApp.java

```diff
@@ -0,0 +1,100 @@
+package com.nytimes.android.sample;
+
+import android.app.Application;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.nytimes.android.external.cache.CacheBuilder;
+import com.nytimes.android.external.fs.SourcePersisterFactory;
+import com.nytimes.android.external.store.base.Persister;
+import com.nytimes.android.external.store.base.impl.BarCode;
+import com.nytimes.android.external.store.base.impl.Store;
+import com.nytimes.android.external.store.base.impl.StoreBuilder;
+import com.nytimes.android.external.store.middleware.GsonParserFactory;
+import com.nytimes.android.sample.data.model.GsonAdaptersModel;
+import com.nytimes.android.sample.data.model.RedditData;
+import com.nytimes.android.sample.data.remote.Api;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import okio.BufferedSource;
+import retrofit2.GsonConverterFactory;
+import retrofit2.Retrofit;
+import retrofit2.RxJavaCallAdapterFactory;
+import rx.Observable;
+
+public class SampleApp extends Application {
+
+    private Store<RedditData, BarCode> nonPersistedStore;
+    private Store<RedditData, BarCode> persistedStore;
+    private Persister<BufferedSource, BarCode> persister;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        initPersister();
+        this.nonPersistedStore = provideRedditStore();
+        this.persistedStore = providePersistedRedditStore();
+    }
+
+    private void initPersister() {
+        try {
+            persister = newPersister();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    public Store<RedditData, BarCode> getNonPersistedStore() {
+        return this.nonPersistedStore;
+    }
+
+    public Store<RedditData, BarCode> getPersistedStore() {
+        return this.persistedStore;
+    }
+
+    private Store<RedditData, BarCode> provideRedditStore() {
+        return StoreBuilder.<RedditData>barcode()
+                .fetcher(barCode -> provideRetrofit().fetchSubreddit(barCode.getKey(), "10"))
+                .memory(CacheBuilder.newBuilder()
+                        .maximumSize(1)
+                        .expireAfterWrite(10, TimeUnit.SECONDS)
```

Comment:
@tsaisean just saw that one. Since @digitalbuddha said he knows what's wrong, I think we should wait till he replies and someone can get on with a fix!